processors/base/writers/location.py

```diff
@@ -70,3 +76,26 @@ def write_location(conn, location, source_id, trial_id=None):
             'created' if create else 'updated', name)
 
     return object['id']
+
+#----helper function to get the canonical name of the locations
+def extract_CanonicalName(location):
+    try:
+        canonical_name = countries.get(location).name
+        logger.debug('Location - %s: %s','normalized', canonical_name)
+    except KeyError:
+        canonical_name = location
+        # This part is for looking for matching cases (suggestion), uses the 
+        # fuzzy logic to get the country name who get 80% of match rate.
+        with open(os.path.join(os.path.dirname(__file__),'countries.csv'), 'r') as f:
+            countries_data = csv.reader(f, delimiter= str(u','))
+            logger.debug('Location - %s: %s','not normalized', canonical_name)
+
+            for country in countries_data:
+                unicode_row = [x.decode('utf8') for x in country]
+                if (fuzz.ratio(location, unicode_row) > 80):
+                    canonical_name = countries.get(country[1]).name
+                    logger.debug('Location - %s: %s','not normalized with the most close result', canonical_name)
```

Comment:
This `for country in countries_data` loop seems not to be doing what I think you'd want to. It goes over every country and, if it's `fuzz.ratio` if over `80`, it saves its name (and keeps going through the loop).

The problem here is that there can be countries that have the ratio, for example, `[60, 100, 85]`. If we're going over them in order, this code would return the name with ratio `85`, even though there's a perfect match in the dataset.